src/circuit_proof/r1cs.rs

```diff
@@ -167,11 +168,30 @@ impl ConstraintSystem {
         }
     }
 
+    // This function can only be called once per ConstraintSystem instance.
     pub fn create_proof_input<R: Rng + CryptoRng>(
-        &self,
+        mut self,
         pedersen_generators: &PedersenGenerators,
         rng: &mut R,
     ) -> (Circuit, ProverInput, VerifierInput) {
+        // If `n`, the number of multiplications, is not 0 or 2, then pad the circuit.
+        let n = self.a.len();
+        if !(n == 0 || n.is_power_of_two()) {
+            let pad_length = n.next_power_of_two() - n;
+            self.a.append(&mut vec![
+                LinearCombination::new(vec![], Scalar::zero());
```

Comment:
Maybe there should be a `LinearCombination::zero()` constructor for this purpose?

Reply:
done!